airflow/executors/base_executor.py

```diff
@@ -287,15 +301,41 @@ def trigger_tasks(self, open_slots: int) -> None:
                 if key in self.attempts:
                     del self.attempts[key]
                 task_tuples.append((key, command, queue, ti.executor_config))
+                s.add_event(
+                    name="task to trigger",
+                    attributes={"command": str(command), "conf": str(ti.executor_config)},
+                )
 
         if task_tuples:
             self._process_tasks(task_tuples)
 
+    @span
     def _process_tasks(self, task_tuples: list[TaskTuple]) -> None:
+        from airflow.traces.utils import gen_span_id_from_ti_key, gen_trace_id
+
         for key, command, queue, executor_config in task_tuples:
-            del self.queued_tasks[key]
-            self.execute_async(key=key, command=command, queue=queue, executor_config=executor_config)
-            self.running.add(key)
+            qt = self.queued_tasks[key][3]
+            trace_id = int(gen_trace_id(qt.dag_run), 16)  # TaskInstance in fourth element
```

Comment:
I think that's a great idea - will apply!

Reply:
So, applying the function with `as_int` boolean did cause problem with pre-commit type checking because it thinks trace_id can now be either str | int. You do end up casting the value with int(..) anyway.

Reply:
oof.    That makes the suggestion less useful... hmm.......  I'll think on it, you shouldn't have to be casting it a million times, there must be a better way.